opfunu/utils/operator.py

```diff
@@ -264,20 +310,23 @@ def hgbat_func(x):
     return np.abs(t2 ** 2 - t1 ** 2) ** 0.5 + (0.5 * t2 + t1) / ndim + 0.5
 
 
+def hgbat_shifted_func(x):
```

Comment:
Please confirm you want parameterized after reading.
I specifically didn't go with a parameterized because I was thinking it would require either an if condition to check the parameter and shift or not, or always perform the shift just with a 0 for some cases if no shift is desired. While small this could introduce some  extra processing. So having a shifted override function which calls the base unshifted seemed like the best performance vs clean approach. 

That said if you feel going with a parameterized approach matches the codebase better I can go what was but would prefer then passing the shift value as 1, -1 etc. with a default of 0. 

Thoughts?

Reply:
@firestrand,

Yes, please set up the parameter name that match with function name. Also set up the default value for it. You don't need to check if else condition here. Your function will be reused more in future. 
Other sections looking good. I will accept this PR after you do it. Thanks

Reply:
@thieu1995 
Started to make the change and it doesn't seem right for the codebase.
Changing the signature to something like `def happy_cat_func(x, shift=0.0)` works fine for when the benchmark is either calling the shifted or unshifted directly. ie. `happy_cat_func(x, shift=-1.0)` or `happy_cat_func(x)`. However with the composite benchmarks the code does something like `self.g2 = operator.happy_cat_func` then in the evaluate we will get `self.g2(mz[self.n1:self.n2], shift=-1.0)` which will be different from the other self.g1,g3 calls.

I considered a functional approach, which would be something like `def happy_cat_func(shift=0.0): def inner(x): <existing logic>` which would then transform the `self.g2 = operator.happy_cat_func(-1.0)` and the evaluate code would be the same. But then the functions that currently call the function directly would look like `operator.happy_cat_func()(x)' which would be different. 

The last alternative is leaving things the way I have with a separate function definition for the shifted case so the signatures in the benchmark stay the same.

So advise option 1, 2, or 3 given this information.

Reply:
@firestrand ,

I forgot to tell you when I defined self.g1 =..., self.g2=... in the __init__ function. It is just for short code. 
Now because you have parameter in the function, then you don't have to define g1, g2,... function.

Just write full function in evaluate(). 
For example, return operator.happy_cat_func((mz[self.n1:self.n2], shift=1.0) + operator.katsuura_func(mz[self.n1:self.n2], shift=-1.0,...) like that.